mailbox/api/src/test/java/org/apache/james/mailbox/events/GroupContract.java

```diff
@@ -275,12 +275,75 @@ default void allGroupListenersShouldBeExecutedWhenGenericGroups() throws Excepti
             verify(listener1, timeout(ONE_SECOND.toMillis()).times(1)).event(any());
             verify(listener2, timeout(ONE_SECOND.toMillis()).times(1)).event(any());
         }
+
+        @Test
+        default void groupListenerShouldReceiveEventWhenRedeliver() throws Exception {
+            MailboxListener listener = newListener();
+
+            eventBus().register(listener, GROUP_A);
+
+            eventBus().reDeliver(GROUP_A, EVENT).block();
+
+            verify(listener, timeout(ONE_SECOND.toMillis()).times(1)).event(any());
+        }
+
+        @Test
+        default void redeliverShouldNotThrowWhenAGroupListenerFails() throws Exception {
```

Comment:
I dont see a test for `redeliver should store event to deadletters when fails to redeliver` it may should be in the deadletter contract

Reply:
FYI : the similar test for dispatch is in `ErrorHandlingContract`, not `EventDeadLettersContract`. So that's where this is gonna be added